File: conformance/tests/post-quantum-slhdsa/post-quantum-slhdsa.runar.zig
const runar = @import("runar");

pub const PostQuantumSLHDSA = struct {
    pub const Contract = runar.SmartContract;

    pubkey: runar.ByteString,

    pub fn init(pubkey: runar.ByteString) PostQuantumSLHDSA {
        return .{ .pubkey = pubkey };
    }

    pub fn spend(self: *const PostQuantumSLHDSA, msg: runar.ByteString, sig: runar.ByteString) void {
        runar.assert(runar.verifySLHDSA_SHA2_128s(msg, sig, self.pubkey));
    }
};

File: conformance/tests/post-quantum-slhdsa/post-quantum-slhdsa.runar.go
//go:build ignore

package contract

import "runar"

type PostQuantumSLHDSA struct {
	runar.SmartContract
	Pubkey runar.ByteString `runar:"readonly"`
}

func (c *PostQuantumSLHDSA) Spend(msg runar.ByteString, sig runar.ByteString) {
	runar.Assert(runar.VerifySLHDSA_SHA2_128s(msg, sig, c.Pubkey))
}
